resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java

```diff
@@ -352,7 +354,7 @@ private void query(final DnsServerAddressStream nameServerAddrStream,
                        final Promise<List<T>> promise,
                        final Throwable cause) {
         if (completeEarly || nameServerAddrStreamIndex >= nameServerAddrStream.size() ||
-                allowedQueries == 0 || promise.isCancelled()) {
+                allowedQueries == 0 || originalPromise.isCancelled() || promise.isCancelled()) {
```

Comment:
@minwoox I wonder if a better approach would be to ensure we propagate cancellation by attaching a `FutureListener` to original promise and call `promise.cancel(...)` if needed. WDYT ?

Reply:
Ah, that was actually my first thought but it makes a little complicated in some cases. For example:
```
1. Call resolve(String inetHost, Promise<T> promise); // The promise here is the original promise
2. Call doResolveUncached() in DnsNameResolver. // A new allPromise is newly created and passed.
3. Call DnsAddressResolveContext.resolve(promise). // The promise here is the allPromise which created in #2.
4. In resolve() searchDomainPromise is created and used.
```
In this flow, there are three `Promises` involved (`originalPromise`, `allPromise` and `searchDomainPromise`), so we have to add two listerers.
So that's the reason that I have chosen this way, but if you think adding listerner is better, then I can change it. 😄

Reply:
@minwoox makes sense.. was just an idea but I guess what you did here is good enough and is less complex.